verilog/analysis/symbol_table_test.cc

```diff
@@ -6132,6 +6132,118 @@ TEST(BuildSymbolTableTest, AnonymousStructTypeNestedMemberReference) {
+                   [](const SymbolTableNode::key_value_type& p) {
+                     return p.first == "idle";
+                   });
+  ASSERT_NE(found_enum_idle, root_symbol.Children().end());
+  const SymbolTableNode& enum_idle(found_enum_idle->second);
+  const SymbolInfo& enum_idle_info(enum_idle.Value());
+  EXPECT_EQ(enum_idle_info.metatype, SymbolMetaType::kTypeAlias);
+  EXPECT_TRUE(enum_idle_info.local_references_to_bind.empty());
+
+  // Find busy symbol
+  const auto found_enum_busy =
+      std::find_if(root_symbol.Children().begin(), root_symbol.Children().end(),
+                   [](const SymbolTableNode::key_value_type& p) {
+                     return p.first == "busy";
+                   });
+  ASSERT_NE(found_enum_busy, root_symbol.Children().end());
+  const SymbolTableNode& enum_busy(found_enum_busy->second);
+  const SymbolInfo& enum_busy_info(enum_busy.Value());
+  EXPECT_EQ(enum_busy_info.metatype, SymbolMetaType::kTypeAlias);
+  EXPECT_TRUE(enum_busy_info.local_references_to_bind.empty());
+
+  // Three references: data and two enum constants
+  ASSERT_EQ(root_symbol.Value().local_references_to_bind.size(), 3);
+
+  // Expect them to bind immediately.
+
+  const ReferenceComponent& anon_enum_ref_comp(
+      root_symbol.Value().local_references_to_bind[2].LastLeaf()->Value());
+  EXPECT_EQ(anon_enum_ref_comp.ref_type, ReferenceType::kImmediate);
+  EXPECT_EQ(anon_enum_ref_comp.required_metatype, SymbolMetaType::kEnumType);
+  EXPECT_EQ(anon_enum_ref_comp.resolved_symbol, &anon_enum);
+
+  const ReferenceComponent& enum_idle_ref_comp(
+      root_symbol.Value().local_references_to_bind[0].LastLeaf()->Value());
+  EXPECT_EQ(enum_idle_ref_comp.ref_type, ReferenceType::kImmediate);
+  EXPECT_EQ(enum_idle_ref_comp.required_metatype,
+            SymbolMetaType::kEnumConstant);
+  EXPECT_EQ(enum_idle_ref_comp.resolved_symbol, &busy);
+
+  const ReferenceComponent& enum_busy_ref_comp(
+      root_symbol.Value().local_references_to_bind[1].LastLeaf()->Value());
+  EXPECT_EQ(enum_busy_ref_comp.ref_type, ReferenceType::kImmediate);
+  EXPECT_EQ(enum_busy_ref_comp.required_metatype,
+            SymbolMetaType::kEnumConstant);
+  EXPECT_EQ(enum_busy_ref_comp.resolved_symbol, &idle);
+
+  // "data"'s type is the (internal) anonymous enum type reference.
+  MUST_ASSIGN_LOOKUP_SYMBOL(data, root_symbol, "data");
+  EXPECT_EQ(data_info.metatype, SymbolMetaType::kDataNetVariableInstance);
+  EXPECT_EQ(data_info.file_origin, &src);
+
+  const DependentReferences& anon_enum_ref(
+      root_symbol.Value().local_references_to_bind[2]);
+  EXPECT_EQ(data_info.declared_type.user_defined_type,
+            anon_enum_ref.LastLeaf());
+
+  {
+    std::vector<absl::Status> resolve_diagnostics;
+    symbol_table.Resolve(&resolve_diagnostics);
+    EXPECT_EMPTY_STATUSES(resolve_diagnostics);
+
+    EXPECT_EQ(anon_enum_ref_comp.resolved_symbol, &anon_enum);  // unchanged
```

Comment:
Can you add the same assertions for busy and idle again here?  Also make sure they are unchanged.

Reply:
Added missing checks.